x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,384 @@
+          : defaultDataViewId ?? '';
+      let initialDataView: DataView[] | undefined;
+      try {
+        const dataView = isESQLQuery(state.query)
+          ? await getESQLAdHocDataview(
+              getIndexPatternFromESQLQuery(state.query.esql),
+              deps.data.dataViews
+            )
+          : await deps.data.dataViews.get(validDataViewId);
+        initialDataView = [dataView];
+      } catch (error) {
+        // Only need to publish blocking error if viewtype is data view, and no data view found
+        if (state.viewType === FieldStatsInitializerViewType.DATA_VIEW) {
+          onError(error);
+        }
+      }
+
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>(initialDataView);
+
+      const subscriptions = new Subscription();
+      if (fieldStatsControlsApi.dataViewId$) {
+        subscriptions.add(
+          fieldStatsControlsApi.dataViewId$
+            .pipe(
+              skip(1),
+              skipWhile((dataViewId) => !dataViewId && !defaultDataViewId),
+              switchMap((dataViewId) => {
+                try {
+                  return deps.data.dataViews.get(dataViewId ?? defaultDataViewId);
+                } catch (error) {
+                  return of(undefined);
+                }
+              })
+            )
+            .subscribe((nextSelectedDataView) => {
+              if (nextSelectedDataView) {
+                dataViews$.next([nextSelectedDataView]);
+              }
+            })
+        );
+      }
+
+      const { toasts } = deps.notifications;
+
+      const api = buildApi(
+        {
+          ...timeRangeApi,
+          ...titlesApi,
+          ...fieldStatsControlsApi,
+          ...dataLoadingApi,
+          getTypeDisplayName: () =>
+            i18n.translate('xpack.dataVisualizer.fieldStats.typeDisplayName', {
+              defaultMessage: 'field statistics',
+            }),
+          isEditingEnabled: () => true,
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableFieldStatsUserInput } = await import(
+                './resolve_field_stats_embeddable_input'
+              );
+              const chartState = serializeFieldStatsChartState();
+              const nextUpdate = await resolveEmbeddableFieldStatsUserInput(
+                coreStart,
+                pluginStart,
+                parentApi,
+                uuid,
+                false,
+                chartState,
+                fieldStatsControlsApi
+              );
+
+              fieldStatsControlsApi.updateUserInput(nextUpdate);
+            } catch (e) {
+              return Promise.reject(e);
+            }
+          },
+          dataViews: dataViews$,
+          serializeState: () => {
+            const dataViewId = fieldStatsControlsApi.dataViewId$?.getValue();
+            const references: Reference[] = dataViewId
+              ? [
+                  {
+                    type: 'index-pattern',
```

Comment:
Switching from `DATA_VIEW_SAVED_OBJECT_TYPE` to `'index-pattern'` will not have an impact on page load bundle size. `x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx` is imported as `const { getFieldStatsChartEmbeddableFactory } = await import('./field_stats_factory');` in x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/index.ts. This means that the file is not included in page load bundle and is only imported when the callback passed to registerReactEmbeddableFactory is called. This callback is not called until the embeddable is rendered in a dashbaord.

Reply:
It was a test that I intended to revert. It's now been reverted.